javalib/src/main/scala/java/net/URI.scala

```diff
@@ -135,13 +135,15 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     case _ => false
   }
 
-  def getAuthority(): String = _authority.map(decodeComponent).orNull
+  def getAuthority(): String =
+    _authority.filter(_ != "").map(decodeComponent).orNull
```

Comment:
Could you please add the examples you used for testing to the tests? As you have certainly observed, the current tests would not expose a behavioral difference between having an empty authority and no authority.